src/core/Akka/Routing/SmallestMailbox.cs

```diff
@@ -35,7 +35,7 @@ private Routee SelectNext(Routee[] routees)
             var winningScore = long.MaxValue;
 
             // round robin fallback
-            var winner = routees[Interlocked.Increment(ref next) % routees.Length];
+            var winner = routees[(Interlocked.Increment(ref next) & int.MaxValue) % routees.Length];
```

Comment:
No sweat @jordansjones - glad that you asked!

When next == Int32.MaxValue the `Interlocked.Increment` operation will cause `next` to "overflow" from a positive to a negative value. In the CLR's normal behavior this will throw an exception because all arithmetic is "checked" for overflows and underflows (subtracting Int32.MinValue to make it positive) by default.

Wrapping this call in an `unchecked` block tells the CLR not to check any arithmetic operations for underflows / overflows and lets the application handle those issues themselves, which we're doing using bitshifting.

So this basically just makes sure that the operation always succeeds.

See the full docs for `unchecked` here https://msdn.microsoft.com/en-us/library/a569z7k8(v=vs.71).aspx